bpf/lib/nodeport.h

```diff
@@ -238,6 +238,73 @@ static __always_inline int dsr_set_ext6(struct __ctx_buff *ctx,
 		return DROP_INVALID;
 	return 0;
 }
+#elif DSR_ENCAP_MODE == DSR_ENCAP_GENEVE
+static __always_inline int encap_geneve_dsr_opt6(struct __ctx_buff *ctx,
+						 struct ipv6hdr *ip6,
+						 const union v6addr *svc_addr,
+						 __be16 svc_port,
+						 int *ifindex, int *ohead)
+{
+	struct remote_endpoint_info *info;
+	union v6addr *dst;
+	bool need_opt = true;
+	__u16 encap_len = sizeof(struct ipv6hdr) + sizeof(struct udphdr) +
+		sizeof(struct genevehdr) + sizeof(struct geneve_dsr_opt6) + ETH_HLEN;
```

Comment:
With TCP we should fine-tune this, and only consider the Option for SYN packets.